### PR TITLE
YD-286 YD-299 Validate email address

### PR DESCRIPTION
**Pull request by Bert-R**
For buddy requests

### CODE REVIEW

File: appservice/src/intTest/groovy/nu/yona/server/AbstractAppServiceIntegrationTest.groovy [hunk 1]
```diff
@@ -95,6 +95,7 @@ abstract class AbstractAppServiceIntegrationTest extends Specification
 	{
 		def richard = addRichard(false)
 		def bob = addBob(false)
+		bob.emailAddress = "bob@dunn.com"
```
Comment (aristotelos) [inline appservice/src/intTest/groovy/nu/yona/server/AbstractAppServiceIntegrationTest.groovy]: Can't this be inlined in `addBob`? That removes a lot of duplicate code that complicates the tests.

File: core/src/main/java/nu/yona/server/subscriptions/service/UserService.java [hunk 2]
```diff
@@ -713,6 +717,15 @@ void assertValidUserFields(UserDto userResource, boolean nickNameIsOptional)
...
 
 		assertValidMobileNumber(userResource.getMobileNumber());
+
+		if (userPurpose == UserPurpose.BUDDY)
+		{
+			if (StringUtils.isBlank(userResource.getEmailAddress()))
+			{
+				throw InvalidDataException.blankEmailAddress();
+			}
+			assertValidEmailAddress(userResource.getEmailAddress());
```
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/subscriptions/service/UserService.java]: Asserting that it is valid can also be done on `UserPurpose.USER` when the e-mail address is not blank.

File: appservice/src/intTest/groovy/nu/yona/server/AbstractAppServiceIntegrationTest.groovy (hunk 1 above)
Comment (Bert-R) [inline appservice/src/intTest/groovy/nu/yona/server/AbstractAppServiceIntegrationTest.groovy]: It's attractive indeed, but cumbersome and error prone. In general, a user does not have an e-mail address, which means that we loose it upon lots of actions: reload, update, number confirmation, etc.

I tried it out, but that won't make things simpler.

File: core/src/main/java/nu/yona/server/subscriptions/service/UserService.java (hunk 2 above)
Comment (Bert-R) [inline core/src/main/java/nu/yona/server/subscriptions/service/UserService.java]: Will do.

**Comment by Bert-R** [comment]
Added the extra check.

**Comment by aristotelos** [comment]
Can be merged now according to me. Should I merge it?

**Comment by Bert-R** [comment]
OK. Let's wait with merging it till the Git problem is resolved. Once that's done, I'll merge it.

**Pull request merged**